src/pageManager.js

```diff
@@ -704,7 +703,15 @@ export async function initialize() {
     }
 
     // Register the pageManager content script for all URLs permitted by the extension manifest.
-    contentScripts.registerContentScript(permissions.getManifestOriginMatchPatterns(), pageManagerContentScript);
+    const matchPatterns = permissions.getManifestOriginMatchPatterns();
+
+    // TODO register this content script so it may be unloaded later. 
+    await browser.scripting.registerContentScripts([{
+        id: "pageManager",
+        js: ["dist/browser-polyfill.min.js", pageManagerContentScript],
+        matches: matchPatterns,
+        persistAcrossSessions: false
```

Comment:
the helper this is replacing had `runAt: document_start` set. It seems that [the default](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/extensionTypes/RunAt) is `document_idle`. Should we keep `document_start`? I wonder if it might make a difference in catching certain events etc?

Reply:
see: https://github.com/mozilla-rally/web-science/blob/043176f033642f6e3d708fab2311aa6b6e3b036a/src/contentScripts.js#L22****

Reply:
Ugh, sorry so apparently I fixed this in https://github.com/mozilla-rally/web-science/pull/302 which is based on this PR, I just cherry-picked that commit over.